projects/ADuCM3029_demo_cn0548/scripts/CN0548_simple_plot.py

```diff
@@ -0,0 +1,206 @@
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#  - The use of this software may or may not infringe the patent rights
+#    of one or more patent holders.  This license does not release you
+#    from the requirement that you obtain separate licenses from these
+#    patent holders to use this software.
+#  - Use of the software either in source or binary form, must be run
+#    on or directly connected to an Analog Devices Inc. component.
+#
+# THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+# IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+# MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+# IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import sys
+import time
+from datetime import datetime
+from typing import List
+import matplotlib.animation as animation
+import matplotlib.pyplot as plt
+import numpy as np
+import adi
+
+while True:
+    mode = input('CN0548 setting: (a/b) \n(a) Low Voltage High Current (absolute max: 15V, 9A)\n(b) High Voltage Low Current (absolute max: 80V, 1.5A)\nMode: ')
+    if mode == 'a' or mode == 'b':
+        break
+    else:
+        print("'a' or 'b' only")
+
+if mode == 'a':
+    v_gain = 0.004
+else:
+    v_gain = 0.02
+i_gain = 0.005
+
+global log
+while True:
+    log = input('\nLog data? (Y/N): ')
+    if log == 'Y' or log == 'N' or log == 'y' or log == 'n':
+        break
+    else:
+        print("'Y' or 'N' only")
+
+if log == 'Y' or log == 'y':
+    global filename
+    dt_now = datetime.now()
+    dt_format = dt_now.strftime("%d-%m-%Y_%H-%M")
+    if mode == 'a':
+        filename = 'LVHC_'+dt_format+'.csv'
+    else:
+        filename = 'HVLC_'+dt_format+'.csv'
+
+
+
+while True:
+    en = input('\nEnable plot? (Y/N): ')
+    if en == 'Y' or en == 'N' or en == 'y' or en == 'n':
+        break
+    else:
+        print("'Y' or 'N' only")
+
+if en == 'Y' or en == 'y':
+    enable_plot = True
+else:
+    enable_plot = False
+
+global fs
+while True:
+    fs = input('\nSamples per second? (1-5): ')
+    try:
+        fs = int(fs)
+    except:
+        print('Please input an integer within the specified range')
+        continue
+    if fs < 1 or fs > 5:
+        print('Please input an integer within the specified range')
+    else:
+        break
+
+global ad7799
+ad7799 = adi.ad7799(uri="serial:COM7,115200")

```

Comment:
can you add a feature also for the COM port number to be passed as input by the user? Similar to the queries above. So that the user can type their own assigned port.

Reply:
Sure, no problem.